source/ShipManager.cpp

```diff
@@ -0,0 +1,190 @@
+/* ShipManager.cpp
+Copyright (c) 2022 by Hurleveur
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "ShipManager.h"
+
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "EsUuid.h"
+#include "GameData.h"
+#include "Messages.h"
+#include "PlayerInfo.h"
+#include "Ship.h"
+
+#include <cstdlib>
+
+using namespace std;
+
+
+
+void ShipManager::Load(const DataNode &node)
+{
+	if(node.Size() < 3 || node.Token(1) != "ship")
+	{
+		node.PrintTrace("Error: Skipping unrecognized node.");
+		return;
+	}
+	taking = node.Token(0) == "take";
+	model = GameData::Ships().Get(node.Token(2));
+	if(node.Size() >= 4)
+		name = node.Token(3);
+
+	for(const DataNode &child : node)
+	{
+		const string &key = child.Token(0);
+		bool hasValue = child.Size() > 1;
+
+		if(key == "id" && hasValue)
+			id = child.Token(1);
+		else if(key == "amount" && hasValue)
+		{
+			int val = child.Value(1);
+			if(val <= 0)
+				child.PrintTrace("Error: \"amount\" must be a non-zero, positive number.");
+			else
+				amount = val;
+		}
+		else if(taking)
+		{
+			if(key == "unconstrained")
+				unconstrained = true;
+			else if(key == "take outfits")
+				takeOutfits = true;
+			else if(key == "requires outfits")
+				requiresOutfits = true;
```

Comment:
If it's going to be `take`, it should be `require`.
If it's going to be `requires`, it should be `takes`.